docs/latest/design/egctl.md

```diff
@@ -0,0 +1,67 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- One-step installation/uninstallation experience
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [subcommand] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `install`, `uninstall`, `config`, `version`.
+
+* `name`: Specifies the name of the specified pod. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-f` or `--file` flags to specify the values for installing.
+
+If you need help, run `kubectl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation   | Syntax                           | Description                                                                 |
+| ----------- | -------------------------------- | --------------------------------------------------------------------------- |
+| `version`   | `egctl version`                  | Prints out build version information.                                       |
+| `install`   | `egctl install -f CUSTOMPROFILE` | Install or reconfigure EG on a cluster.                                     |
+| `uninstall` | `egctl uninstall`                | Uninstall EG from a cluster                                                 |
+| `config`    | `egctl config SUBCOMMAND`        | Retrieve information about proxy configuration from envoy proxy and gateway |
+| `analyze`   | `egctl analyze`                  | Analyze configuration and print validation messages                         |
+
+## Examples
+
+Use the following set of examples to help you familiarize yourself with running the commonly used `egctl` operations:
+
+```console
+# Install EG using the definition in custom profile file
+egctl install -f custom-profile.yaml
```

Comment:
there is no notion of custom profile in EG but a config file, also can we follow same convention of using `-c` https://github.com/envoyproxy/gateway/blob/46b7b7523cc8d8167fd999c9cdecdca2ffb9fd71/internal/cmd/server.go#L36

Reply:
Note that if we intend on egctl eventually installing the data-plane, the `-c` will need to support the EnvoyGateway and EnvoyProxy APIs. For `egctl install -c example.yml`, example.yml will be as follows:
```
apiVersion: config.gateway.envoyproxy.io/v1alpha1
kind: EnvoyGateway
provider:
  type: Kubernetes
gateway:
  controllerName: gateway.envoyproxy.io/gatewayclass-controller
---
apiVersion: config.gateway.envoyproxy.io/v1alpha1
kind: EnvoyProxy
metadata:
  name: example-config
spec:
  provider:
    type: Kubernetes
    kubernetes:
      envoyDeployment:
        replicas: 3
```

__Notes:__
- The example does not specify `control-plane` or `data-plane` so both will be installed.
- If the yaml referenced by `-c` does not specify EnvoyProxy, then the Envoy proxy infra is created with default settings.
- A GatewayClass with a `spec.controllerName` value that matches EG's controllerName should be created when the `control-plane` is installed.
- A Gateway with a `spec.gatewayClassName` value that matches the above GatewayClass should be created when the `data-plane` is installed.

As I mentioned previously, `egctl` can start off by only managing the control-plane. I provide the data-plane examples and details as design considerations.